### PR TITLE
Updated proto files, added layers to caffe

### LINKED ISSUE #912: Opencv_dnn >> can't load network ResNet-101 for face recogniiton
##### System information (version)
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015

##### Detailed description

Hello. I have found that ResNet-101 cannot be loaded by opencv dnn module. When try to load this message appears:
```
[libprotobuf ERROR C:\Programming\3rdParties\opencv310\opencv_contrib\modules\dn
n\3rdparty\protobuf\sources\protobuf-3.1.0\src\google\protobuf\text_format.cc:29
8] Error parsing text-format caffe.NetParameter: 33:26: Message type "caffe.Laye
rParameter" has no field named "batch_norm_param".

OpenCV Error: Unspecified error (FAILED: ReadProtoFromTextFile(param_file, param
). Failed to parse NetParameter file: ResNet-101-deploy_augmentation.prototxt) in cv::dnn::ReadNetParamsFromTextFileOrDie, file C:\Programming\3rdParties\opencv310\opencv_contrib\mo
dules\dnn\src\caffe\caffe_io.cpp, line 1101 C:\Programming\3rdParties\opencv310\opencv_contrib\modules\dnn\src\caffe\caffe_io.cpp:1101: error: (-2) FAILED: ReadProtoFromTextFile(param_file, param). Failed to parse NetParameter file: ResNet-101-deploy_augmentation.prototxt in function cv::dnn::ReadNetParamsFromTextFileOrDie
```

##### Steps to reproduce
1) Download [ResNet-101](https://docs.google.com/forms/d/e/1FAIpQLSdterS7LCr2hVb-MJWhbdI6AgDDvN0qL45CptoGCbFMbt1F8g/viewform)

2) Change prototxt and caffemodel file names into this [example](https://github.com/opencv/opencv_contrib/blob/master/modules/dnn/samples/caffe_googlenet.cpp)

3) Build and execute

### PR DESCRIPTION
**Pull request by arrybn**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/323

resolves #912 
### This pullrequest changes
Updated caffe models importer, so it can read models with BatchNormalization layer. Also added some tests

### CODE REVIEW

**Comment by young-noul** [comment]
I'm looking forward to use this commit in the master branch ASAP :)
It solves a really big issue using batch normalization layer that is now essential in DNN architectures.

**Comment by alalek** [comment]
:+1:
